### PR TITLE
ESLint Flat Config

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Convert the eslint lint and format to use the flat config configuration. (The current .eslint-rc configuration is deprecated in eslint 9.)
<s>Still awaits an eslint release so we don't need to use the latest eslint dev.</s>

## Type of change

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -256,7 +256,8 @@
     "lit": "2.8.0",
     "clean-css": "5.3.3",
     "@lit/reactive-element": "1.6.3",
-    "@fullcalendar/daygrid": "6.1.15"
+    "@fullcalendar/daygrid": "6.1.15",
+    "globals": "^13.0.0"
```
Comment (bramkragten) [inline package.json:260]: Why is globals added as a resolution?

File: package.json [hunk 2]
```diff
@@ -196,12 +196,12 @@
     "browserslist-useragent-regexp": "4.1.3",
     "chai": "5.1.1",
     "del": "7.1.0",
-    "eslint": "8.57.1",
+    "eslint": "eslint/eslint",
```
Comment (bramkragten) [inline package.json:199]: 🤔 weird version number

File: package.json (hunk 1 above)
Comment (silamon) [reply package.json:260]: Older versions have a space mistake where eslint now starts to fail. We still throw in version 11 somewhere.

File: package.json (hunk 2 above)
Comment (silamon) [reply package.json:199]: Yes, awaiting a new release somewhere next week. Now it points to their github repo and takes in latest dev.

Comment (silamon) [reply package.json]: The new version has been released (9.12.0) so this is now replaced.

**Comment by silamon** [comment]
Merging this will book progress in both #20762 and #21574.

**Review by wendevlin — DISMISSED** [review]
Thank you :tada: 
@silamon can you please resolve the merge conflict?

**Review by wendevlin — APPROVED** [review]

**Pull request merged**